lib/rspec/support/source/node.rb

```diff
@@ -0,0 +1,244 @@
+          :alias,
+          :alias_error,
+          :aref,
+          :aref_field,
+          :arg_ambiguous,
+          :arg_paren,
+          :args_add,
+          :args_add_block,
+          :args_add_star,
+          :args_new,
+          :array,
+          :assign,
+          :assign_error,
+          :assoc_new,
+          :assoc_splat,
+          :assoclist_from_args,
+          :bare_assoc_hash,
+          :begin,
+          :binary,
+          :block_var,
+          :block_var_add_block,
+          :block_var_add_star,
+          :blockarg,
+          :bodystmt,
+          :brace_block,
+          :break,
+          :call,
+          :case,
+          :class,
+          :class_name_error,
+          :command,
+          :command_call,
+          :const_path_field,
+          :const_path_ref,
+          :const_ref,
+          :def,
+          :defined,
+          :defs,
+          :do_block,
+          :dot2,
+          :dot3,
+          :dyna_symbol,
+          :else,
+          :elsif,
+          :ensure,
+          :excessed_comma,
+          :fcall,
+          :field,
+          :for,
+          :hash,
+          :heredoc_dedent,
+          :if,
+          :if_mod,
+          :ifop,
+          :lambda,
+          :magic_comment,
+          :massign,
+          :method_add_arg,
+          :method_add_block,
+          :mlhs_add,
+          :mlhs_add_star,
+          :mlhs_new,
+          :mlhs_paren,
+          :module,
+          :mrhs_add,
+          :mrhs_add_star,
+          :mrhs_new,
+          :mrhs_new_from_args,
+          :next,
+          :opassign,
+          :operator_ambiguous,
+          :param_error,
+          :params,
+          :paren,
+          :parse_error,
+          :program,
+          :qsymbols_add,
+          :qsymbols_new,
+          :qwords_add,
+          :qwords_new,
+          :redo,
+          :regexp_add,
+          :regexp_literal,
+          :regexp_new,
+          :rescue,
+          :rescue_mod,
+          :rest_param,
+          :retry,
+          :return,
+          :return0,
+          :sclass,
+          :stmts_add,
+          :stmts_new,
+          :string_add,
+          :string_concat,
+          :string_content,
+          :string_dvar,
+          :string_embexpr,
+          :string_literal,
+          :super,
+          :symbol,
+          :symbol_literal,
+          :symbols_add,
+          :symbols_new,
+          :top_const_field,
+          :top_const_ref,
+          :unary,
+          :undef,
+          :unless,
+          :unless_mod,
+          :until,
+          :until_mod,
+          :var_alias,
+          :var_field,
+          :var_ref,
+          :vcall,
+          :void_stmt,
+          :when,
+          :while,
+          :while_mod,
+          :word_add,
+          :word_new,
+          :words_add,
+          :words_new,
+          :xstring_add,
+          :xstring_literal,
+          :xstring_new,
+          :yield,
+          :yield0,
+          :zsuper
+        ]
```

Comment:
I worry that this array will get out of sync with the ripper constant.  It looks like it's only needed to dynamically define all the `*_type?` methods.  Instead, WDYT about defining  a single `of_type?(event)` method or just do manual `==` comparisons?  Then you wouldn't need this constant, this class would be smaller/simpler, and we wouldn't have to worry about the constant going out of sync with Ripper's.

On a side note: the rspec-core `Node` class did not need this, so we do we need it now?